src/main/java/nextstep/courses/service/SessionService.java

```diff
@@ -0,0 +1,33 @@
+package nextstep.courses.service;
+
+import nextstep.courses.domain.Session;
+import nextstep.courses.domain.SessionRepository;
+import nextstep.payments.domain.Payment;
+import nextstep.payments.service.PaymentService;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class SessionService {
+    private final SessionRepository sessionRepository;
+    private final PaymentService paymentService;
+
+    public SessionService(SessionRepository sessionRepository, PaymentService paymentService) {
+        this.sessionRepository = sessionRepository;
+        this.paymentService = paymentService;
+    }
+
+    @Transactional
+    public void enroll(Long sessionId, Long userId, String paymentId) {
+        Session session = sessionRepository.findById(sessionId)
+                .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 강의입니다."));
+
+        Payment payment = null;
+        if (session.isPaid()) {
+            payment = paymentService.payment(paymentId);
+        }
+
+        session.enroll(payment);
```

Comment:
수강을 신청한 수강생 정보도 신청할 때 필요하지 않을까요?
```suggestion
        session.enroll(user, payment);
```

Reply:
요구사항에 구체적으로 제시된 기능은 아니지만 필요하겠네요. 수정했습니다